### PR TITLE
Add AngularStrap type definitions

### PR DESCRIPTION
**Pull request by samherrmann**
## TypeScript definitions for AngularStrap

http://mgcrea.github.io/angular-strap/
#### General Comments

For anyone reviewing this pull request, you should know that I've had minimal hands-on experience with AngularStrap. There are a number of areas where I was unsure what the type definition should be. Any feedback is therefore welcome!

All tests I wrote use the examples from the AngularStrap website. Most modules only have an example for how to use the provider though, meaning that the services for most modules don't have a test at the moment. I found that there is very little documentation about services. I dug into the source code for most of them and tried my best to define all of the members that I think are meant for use in an app.

My aim is to have this reviewed by at least one other person who is a little more familiar with AngularStrap before I create a pull request over at DefinitelyTyped.

### CODE REVIEW

File: angular-strap/angular-strap.d.ts [hunk 1]
```diff
@@ -0,0 +1,600 @@
...
+        }
+    
+        interface IDatepickerProvider {
+            defaults: IDatepickerOptions;
+        }
+    
+        interface IDatepicker {
+            update: (date: Date) => void;      
+            updateDisabledDates: (dateRanges: IDatepickerDateRange[]) => void;     
+            select: (dateConstructorArg: string | number | number[], keep: boolean) => void;
```
Comment (samherrmann) [inline angular-strap/angular-strap.d.ts:302]: is _keep_ actually a _boolean_?

File: angular-strap/angular-strap.d.ts [hunk 2]
```diff
@@ -0,0 +1,600 @@
...
+            setMode: (mode: any) => void;
```
Comment (samherrmann) [inline angular-strap/angular-strap.d.ts:303]: what type should _mode_ be?

File: angular-strap/angular-strap.d.ts [hunk 3]
```diff
@@ -0,0 +1,600 @@
...
+        interface ISelectService {
+            (element: ng.IAugmentedJQuery, controller: any, config: ISelectOptions): ISelect;
+        }
+        
+        interface ISelectProvider {
+            defaults: ISelectOptions;
+        }
+        
+        interface ISelect {
+            update: (matches: any) => void;
```
Comment (samherrmann) [inline angular-strap/angular-strap.d.ts:413]: what type is _matches_?

File: angular-strap/angular-strap.d.ts [hunk 4]
```diff
@@ -0,0 +1,600 @@
...
+            defaults: IScrollspyOptions;
+        }
+        
+        interface IScrollspyService {
+            (element: ng.IAugmentedJQuery, options: IScrollspyOptions): IScrollspy;
+        }
+        
+        interface IScrollspy {
+            checkOffsets: () => void;
+            trackElement: (target: any, source: any) => void;
```
Comment (samherrmann) [inline angular-strap/angular-strap.d.ts:559]: what type should _target_ and _source_ be?

File: angular-strap/angular-strap-tests.ts [hunk 5]
```diff
@@ -0,0 +1,378 @@
+/// <reference path="../angularjs/angular.d.ts"/>
```
Comment (samherrmann) [inline angular-strap/angular-strap-tests.ts]: ### General Comments

For anyone reviewing this pull request, you should know that I've had minimal hands-on experience with AngularStrap. There are a number of areas where I was unsure what the type definition should be. Any feedback is therefore welcome!

All tests I wrote use the examples from the AngularStrap website. Most modules only have an example for how to use the provider though, meaning that the services for most modules don't have a test at the moment. I found that there is very little documentation about services. I dug into the source code for most of them and tried my best to define all of the members that I think are meant to be public.

**Comment by stevehipwell** [comment]
What is the state of progress on this pull request? Surely having a definition is better than not having one?

**Pull request closed without merging**